### PR TITLE
sui: add example message app

### LINKED ISSUE #2529: sui: example contract publishing core message
We should duplicate the useful `core_messages` example [app](https://github.com/wormhole-foundation/wormhole/tree/main/aptos/examples/core_messages) that we have in the `aptos` dir for Sui. This provides guidance for integrators and also lets us maintain the functionality that might be lost because of #2528 by having the `worm` CLI call this example message publish fn.

### PR DESCRIPTION
**Pull request by heyitaki**
This PR introduces an example app as a simple demonstration of sending messages via the core bridge. It also restores the ability to test via the `publish_message.sh` script.

To use `publish-message` CLI command, you will need the output of `scripts/deploy`:

For the command `worm sui publish-message -n devnet -p <package-id> -s <example-app-state-object-id> -w <wormhole-state-object-id> -m "hello"`:
* `package-id`: output that follows `Deployed to: ` in the `[+1]` step in output from `scripts/deploy`
* `example-app-state-object-id`: output that follows `Example app state object ID:` in the `[+1]` step
* `wormhole-state-object-id`: output that follows `Wormhole state object ID:` in step `[3/4]`

Resolves #2529

### CODE REVIEW

**Comment by heyitaki** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2358** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2358" target="_blank">Graphite</a> 
    * **PR #2516** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2516" target="_blank">Graphite</a> 
    * **PR #2494** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2494" target="_blank">Graphite</a> 
    * **PR #2543** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2543" target="_blank">Graphite</a> 
      * **PR #2544** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2544" target="_blank">Graphite</a> 
        * **PR #2534** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2534" target="_blank">Graphite</a> 
          * **PR #2547** <a href="https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2547" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2547?utm_source=stack-comment).

File: sui/examples/core_messages/sources/sender.move [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    }
+
+    fun init(ctx: &mut TxContext) {
+        transfer::transfer(
+            DeployerCap { id: object::new(ctx) },
+            tx_context::sender(ctx)
+        );
+    }
+
+    public entry fun init_with_params(
```
Comment (kcsongor) [inline sui/examples/core_messages/sources/sender.move:27]: the `DeployerCap` pattern is used elsewhere to ensure that only the deployer can initialise the contract with the right parameters (this protects against a malicious DoS attack). However, in the case of this example, the only external parameter used is the reference to the `WormholeState` object, which is globally unique anyway, so I think we should simplify the example by dropping the `DeployerCap` stuff.

File: sui/examples/core_messages/sources/sender.move [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        let fee_coins = coin::split(
+            &mut fee_with_potential_surplus,
+            fee_amount,
+            ctx
+        );
+        transfer::transfer(fee_with_potential_surplus, tx_context::sender(ctx));
+        wormhole::publish_message::publish_message(
+            wormhole_state,
+            &mut state.emitter_cap,
+            0, // Set nonce to 0, only used for batch VAAs.
```
Comment (kcsongor) [inline sui/examples/core_messages/sources/sender.move:78]: maybe just say the nonce is (currently) unused

**Review by kcsongor — APPROVED** [review]

**Comment by kcsongor** [comment]
nice!

**Comment by heyitaki** [comment]
@heyitaki queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2547).

**Comment by heyitaki** [comment]
[Graphite](https://app.graphite.dev/github/pr/wormhole-foundation/wormhole/2547) couldn't merge this pull request because a downstack PR #2534 failed to merge.

**Pull request merged**